libr/core/canal.c

```diff
@@ -725,30 +725,32 @@ static int core_anal_fcn(RCore *core, ut64 at, ut64 from, int reftype, int depth
 	if (!fcn->name) {
 		fcn->name = r_str_newf ("%s.%08"PFMT64x, fcnpfx, at);
 	}
-	buflen = core->anal->opt.bb_max_size;
-	buf = calloc (1, buflen);
-	if (!buf) {
-		eprintf ("Error: malloc (buf)\n");
-		goto error;
-	}
 	do {
 		RFlagItem *f;
 		int delta = r_anal_fcn_size (fcn);
 		// XXX hack slow check io error
 		if (core->io->va) {
-			if (!r_io_is_valid_offset (core->io, at+delta, !core->anal->opt.noncode)) {
+			if (!r_io_is_valid_offset (core->io, at + delta, !core->anal->opt.noncode)) {
+				goto error;
+			}
+		} else {
+			if (!r_io_is_valid_offset (core->io, at + delta, !core->anal->opt.noncode)) {
```

Comment:
no need for the `else`, this is the same code as the one in the `if` block, so just do it always, or leave it as it was before (only done in the `if`).